### PR TITLE
Make cra-kitchen-sink test snapshot locale-independent

### LINKED ISSUE #1521: Tests are failing locally
With this setup:
```
$ git clone https://github.com/storybooks/storybook.git
$ cd storybook
$ yarn
$ yarn bootstrap
```

if you run `yarn test`, it just fails to start:
```
$ yarn test
yarn test v0.27.5
$ jest --projects ./ ./examples/react-native-vanilla

error Command failed with exit code 1.
```

`yarn jest` works, but one test is failing: 
```
Summary of all failing tests
 FAIL  examples/cra-kitchen-sink/src/storyshots.test.js
  ● Storyshots › Button › with knobs

    expect(value).toMatchSnapshot()

    Received value does not match stored snapshot 1.

    - Snapshot
    + Received

    @@ -10,11 +10,11 @@
       <p>
         My name is Storyteller, I'm 70 years old, and my favorite fruit is apple.
       </p>
       <p>
         My birthday is:
    -    2017-1-20
    +    1/20/2017
       </p>
       <p>
         My wallet contains: $
         12.50
       </p>
```

I believe the second issue is connected with usage of [toLocaleDateString](https://github.com/storybooks/storybook/blob/master/examples/cra-kitchen-sink/src/stories/index.js#L81)

### PR DESCRIPTION
**Pull request by Hypnosphi**
Fixes #1521

### CODE REVIEW

File: examples/cra-kitchen-sink/src/stories/index.js [hunk 1]
```diff
@@ -78,7 +78,7 @@ storiesOf('Button', module)
           {intro}
         </p>
         <p>
-          My birthday is: {new Date(birthday).toLocaleDateString()}
+          My birthday is: {new Date(birthday).toDateString()}
```
Comment (danielduan) [inline examples/cra-kitchen-sink/src/stories/index.js]: This is still locale dependant unfortunately.

I think the solution would be to keep using `toLocaleDateString` but pass in a locale and a format option.

```
var options = { weekday: 'short', year: 'numeric', month: 'short', day: 'numeric' };
date.toLocaleDateString('en-US', options);
```

or something like this.

**Review by danielduan — CHANGES_REQUESTED** [review]

**Comment by ndelangen** [comment]
@Hypnosphi I've previously fixed this in a similar way (see merge conflicts). Thank you for helping out!! If you can find any issues related to this, please let me know or reopen and rebase this PR?

Thank you!

**Comment by Hypnosphi** [comment]
@ndelangen looks like your change solves the issue only for us europeans =)

**Comment by Hypnosphi** [comment]
@ndelangen looks like your change solves the issue only for us europeans, people from USA should get error now (I wonder why CI passes though)

**Review by shilman — APPROVED** [review]
LGTM!

**Pull request merged**